Library/OcCryptoLib/Sha2-avx.h

```diff
@@ -0,0 +1,92 @@
+/** @file
+  Copyright (C) 2021, ISP RAS. All rights reserved.
+
+  All rights reserved.
+
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+**/
+
+#ifndef SHA2_AVX_H
+#define SHA2_AVX_H
+
+#include <Library/OcDebugLogLib.h>
+#include <Library/OcCryptoLib.h>
+
+#define UNPACK64(x, str)                         \
+  do {                                           \
+    *((str) + 7) = (UINT8) (x);                  \
+    *((str) + 6) = (UINT8) RShiftU64 ((x),  8);  \
+    *((str) + 5) = (UINT8) RShiftU64 ((x), 16);  \
+    *((str) + 4) = (UINT8) RShiftU64 ((x), 24);  \
+    *((str) + 3) = (UINT8) RShiftU64 ((x), 32);  \
+    *((str) + 2) = (UINT8) RShiftU64 ((x), 40);  \
+    *((str) + 1) = (UINT8) RShiftU64 ((x), 48);  \
+    *((str) + 0) = (UINT8) RShiftU64 ((x), 56);  \
+  } while(0)
+
+#define PACK64(str, x)                           \
+  do {                                           \
+    *(x) =    ((UINT64) *((str) + 7))            \
+           | LShiftU64 (*((str) + 6),  8)        \
+           | LShiftU64 (*((str) + 5), 16)        \
+           | LShiftU64 (*((str) + 4), 24)        \
+           | LShiftU64 (*((str) + 3), 32)        \
+           | LShiftU64 (*((str) + 2), 40)        \
+           | LShiftU64 (*((str) + 1), 48)        \
+           | LShiftU64 (*((str) + 0), 56);       \
+  } while (0)
+
+typedef struct {
+  UINT64 State[8];
+  UINT64 Length;
+  UINT64 TotalLength;
+  UINT8  Block[2 * SHA512_BLOCK_SIZE];
+} SHA512_STATE;
+
+typedef SHA512_STATE SHA384_STATE;
+
+//
+// Sha 384 Init State
+//
+STATIC CONST UINT64 SHA384_H0[8] = {
```

Comment:
In my opinion, better make these declarations externs, i.e. `extern CONST UINT64 SHA384_H0[8];` and provide the definition in a separate C file. That will reduce the resulting binary size by avoiding the duplication of these arrays. While not by much, I believe it is worth it.